### PR TITLE
refactor: drop lodash some, castArray and omitBy

### PR DESCRIPTION
**Pull request by coreyfarrell**
I am thinking lodash.camelCase and lodash.kebabCase should probably be replaced with camelcase and decamelize which is used by yargs-parser.  Should be possible to use `is-plain-obj` or `is-plain-object` which would eliminate lodash.

### CODE REVIEW

**Comment by bcoe** [comment]
@coreyfarrell for consistency, would be happy to switch to camelcase and decamelize, seems correct to use the same libraries in both contexts; want to update this PR to reflect this?

**Comment by coreyfarrell** [comment]
@bcoe sure.  any thoughts about using `is-plain-obj` or `is-plain-object` to eliminate lodash entirely?  maybe do that separate?

**Comment by bcoe** [comment]
@coreyfarrell I'm comfortable with this, in this PR. Mainly, because we won't be using many of lodash's features, and it creates more parity between yargs and yargs-unparser. Usually, for this variety of library, I'm pretty comfortable with having `lodash` as a dependendency:

* it's extremely well tested.
* used by many many folks.
* etc.

tldr; love the argument, let's make our two libraries more similar, the arguments of let's reduce our install footprint by a few bytes don't usually land with me.

**Comment by coreyfarrell** [comment]
lodash is 4.9MB installed.  I can see your argument when talking about development only tools, not so much for modules that might be used in production.

**Comment by coreyfarrell** [comment]
CC @cspotcode.  This does add a few more `require` files back in but decamelize and camelcase are both already required via yargs, so effectively we're replacing lodash with is-plain-obj (261 bytes instead of 528k).

**Comment by cspotcode** [comment]
What's the difference in startup time?  Do you have a benchmark we can paste into our terminals?

Arguments about install size aren't compelling; a few MB here and there really doesn't matter, especially when you consider the machines on which this'll be running, and the ways it'll be used.

**Comment by coreyfarrell** [comment]
I just ran a benchmark, on my system it seems like this PR makes the startup even faster.  I ran the benchmark from the branch for this PR, first I ran `git show origin/master:index.js > old.js` to retrieve the existing script.

```sh
#!/usr/bin/env sh

for i in 1 2 3 4 5 6 7 8 9 10 ; do
  node -e 'console.time(process.argv[1]);require(process.argv[1]);console.timeEnd(process.argv[1]);' ./index.js
done

for i in 1 2 3 4 5 6 7 8 9 10 ; do
  node -e 'console.time(process.argv[1]);require(process.argv[1]);console.timeEnd(process.argv[1]);' ./old.js
done
```

Results:
```
./index.js: 14.104ms
./index.js: 11.520ms
./index.js: 14.188ms
./index.js: 11.292ms
./index.js: 11.490ms
./index.js: 10.474ms
./index.js: 14.603ms
./index.js: 11.258ms
./index.js: 10.797ms
./index.js: 11.410ms
./old.js: 22.837ms
./old.js: 19.579ms
./old.js: 24.166ms
./old.js: 19.842ms
./old.js: 22.665ms
./old.js: 19.674ms
./old.js: 22.352ms
./old.js: 19.564ms
./old.js: 19.696ms
./old.js: 20.116ms
```
./index.js is from this PR, ./old.js is from the current master.

**Comment by coreyfarrell** [comment]
I just tested this using code from before #25 was merged, in that case each iteration took about 34ms.  I'd be interested in knowing what results you see.  My system has SSD storage so I don't want to ignore the possibility that files resolve and load abnormally fast for me.

**Comment by cspotcode** [comment]
Sweet, I see similar results on Linux, and it looks faster on Windows as well.  (also SSD here)
